tests/integration_tests/conftest.py

```diff
@@ -92,7 +92,11 @@ def session_cloud() -> Generator[IntegrationCloud, None, None]:
     cloud = platforms[integration_settings.PLATFORM](image_type=image_type)
     cloud.emit_settings_to_log()
     yield cloud
-    cloud.destroy()
+
+    if not integration_settings.KEEP_INSTANCE or (
```

Comment:
We're already doing the `KEEP_INSTANCE` check in `destroy()` of the cloud, so we should probably move this new check there.

Also, we need to update https://github.com/canonical/cloud-init/blob/c9b9eef30252f6470e0b132d831ceb4de2521c38/tests/integration_tests/instances.py#L327